commands.py

```diff
@@ -920,68 +898,51 @@ def _get_tracking_courses(c_handler: CanvasHandler, CANVAS_API_URL) -> discord.E
-                        since = re.sub(r"\s", "-", since)
-                        data_list = ch.get_course_stream_ch(since, (str(c.id),), CANVAS_API_URL, CANVAS_API_KEY)
+                for c in ch.courses:
+                    since = ch.timings[str(c.id)]
+                    since = re.sub(r"\s", "-", since)
+                    data_list = ch.get_course_stream_ch(since, (str(c.id),), CANVAS_API_URL, CANVAS_API_KEY)
 
-                        for data in data_list:
-                            embed_var = discord.Embed(title=data[2], url=data[3], description=data[4], color=CANVAS_COLOR)
-                            embed_var.set_author(name=data[0], url=data[1])
-                            embed_var.set_thumbnail(url=CANVAS_THUMBNAIL_URL)
-                            embed_var.add_field(name="Created at", value=data[5], inline=True)
+                    for data in data_list:
+                        embed_var = discord.Embed(title=data[2], url=data[3], description=data[4], color=CANVAS_COLOR)
+                        embed_var.set_author(name=data[0], url=data[1])
+                        embed_var.set_thumbnail(url=CANVAS_THUMBNAIL_URL)
+                        embed_var.add_field(name="Created at", value=data[5], inline=True)
 
-                            for channel in ch.live_channels:
-                                await channel.send(notify_role.mention if notify_role else "", embed=embed_var)
+                        for channel in ch.live_channels:
+                            await channel.send(notify_role.mention if notify_role else "", embed=embed_var)
 
-                        if data_list:
-                            # latest announcement first
-                            ch.timings[str(c.id)] = data_list[0][5]
+                    if data_list:
+                        # latest announcement first
+                        ch.timings[str(c.id)] = data_list[0][5]
 
             await asyncio.sleep(3600)
 
     @staticmethod
     async def assignment_reminder(self):
         while True:
-            for ch in self.d_handler.canvas_handlers:
-                if ch.live_channels:
-                    notify_role = None
+            for ch in filter(operator.attrgetter("live_channels"), self.d_handler.canvas_handlers):
+                notify_role = next((r for r in ch.guild.roles if r.name.lower() == "notify"), None)
 
-                    for role in ch.guild.roles:
-                        if role.name == "notify":
-                            notify_role = role
-                            break
-
-                    for c in ch.courses:
-                        data_list = ch.get_assignments("1-week", (str(c.id),), CANVAS_API_URL)
+                for c in ch.courses:
+                    for time in ("week", "day"):
+                        data_list = ch.get_assignments(f"1-{time}", (str(c.id),), CANVAS_API_URL)
                         recorded_ass_ids = ch.due_week[str(c.id)]
```

Comment:
the variable naming here is top notch, @Vismai-Khanderao

Reply:
_p y t h o n i c_